File: src/main/java/app/askresume/global/model/ApiResult.java
package app.askresume.global.model;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.springframework.http.HttpStatus;

import java.time.LocalDateTime;

@Getter
@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class ApiResult<T> {

    private final LocalDateTime timestamp = LocalDateTime.now();

    @Schema(description = "status", example = "200")
    private final int status = HttpStatus.OK.value();
    private T data;

    public ApiResult(T data) {
        this.data = data;
    }
}

File: src/main/kotlin/app/askresume/global/model/ApiResult.kt
package app.askresume.global.model

import com.fasterxml.jackson.annotation.JsonFormat
import io.swagger.v3.oas.annotations.media.Schema
import org.springframework.http.HttpStatus
import java.time.LocalDateTime

data class ApiResult<T>(
    val data: T,
) {

    @field:JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd HH:mm:ss", timezone = "Asia/Seoul")
    val timestamp: LocalDateTime = LocalDateTime.now()

    @Schema(description = "status", example = "200")
    val status = HttpStatus.OK.value()
}